src/PlaywrightSharp/Helpers/FlexibleStringEnumConverter.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace PlaywrightSharp.Helpers
+{
+    internal class FlexibleStringEnumConverter<T> : JsonConverter<T>
```

Comment:
I think this should be covered by specific UT's to test the scenarios it supports

Reply:
@Meir017 we already have unit tests checking for resource types. I don't know if I would want to leave `images` and `beacon` out of the enum now that we know about them...

Reply:
I meant to test `FlexibleStringEnumConverter` in general. not playwright related flows ;)

Reply:
That's a good one!